src/marshmallow/orderedset.py

```diff
@@ -70,6 +70,15 @@ def pop(self, last=True):
         self.discard(key)
         return key
 
+    def union(self, other):
+        return self | other
```

Comment:
This isn't quite correct, or at least it's inconsistent with regular `set`s because it doesn't cast `other`.

https://til.sloria.com/#2018-06-13-python-sets-difference-between-a-or-b-and-aunionb

Reply:
Oh you're right.